PR for release 1.1.0

KevinMenden opened the pull request:
PR for the release of version 1.1.0.

This release brings a few major and potentially breaking changes.
The main ones:
* `--input`  instead of `--reads` for defining the input data
* `nfcore_schema.json` file for parameters
* `lib` folder with groovy helper functions

KevinMenden commented:
Small things I just found:

* no default value for `single_end` parameter (we could also remove that completely as not supported?)
* some unexpected params

Will fix those and anything else I find in a separate PR

ewels commented:
Not sure why the linting markdown comment didn't make it, maybe because the PR is to `master` and the comment stuff needs to be there first. Anyway, here it is:

---

## `nf-core lint` overall result: Passed :white_check_mark: :warning:

Posted for pipeline commit 0b54f7b

```diff
+| ✅ 205 tests passed       |+
!| ❗  20 tests had warnings |!
```

<details>

### :heavy_exclamation_mark: Test warnings:

* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config ``manifest.version`` should end in ``dev``: ``'1.1.0'``
* [actions_awsfulltest](https://nf-co.re/tools-docs/lint_tests/actions_awsfulltest.html) - `.github/workflows/awsfulltest.yml` should test full datasets, not `-profile test`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `conda-forge::r-base=3.6.3`, `4.0.5` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `conda-forge::r-statmod=1.4.32`, `1.4.36` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `conda-forge::r-data.table=1.12.4`, `1.14.0` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `conda-forge::r-gplots=3.0.1.1`, `3.1.1` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `conda-forge::r-r.methodss3=1.7.1`, `1.8.1` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `conda-forge::python=3.7.3`, `3.9.4` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `conda-forge::markdown=3.1.1`, `3.3.4` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `conda-forge::pymdown-extensions=6.0`, `8.2` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `conda-forge::pygments=2.6.1`, `2.9.0` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `conda-forge::pigz=2.3.4`, `2.6` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `bioconda::trim-galore=0.6.5`, `0.6.6` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `bioconda::samtools=1.9`, `1.12` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `bioconda::bowtie=1.2.3`, `1.3.0` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `bioconda::multiqc=1.9`, `1.10.1` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `bioconda::htseq=0.11.3`, `0.13.5` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `bioconda::seqkit=0.12.0`, `0.16.0` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `bioconda::bioconductor-edger=3.26.5`, `3.32.1` available
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep outdated: `bioconda::bioconductor-limma=3.40.2`, `3.46.0` available

### :white_check_mark: Tests passed:

* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.gitattributes`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `CHANGELOG.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `CODE_OF_CONDUCT.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `CODE_OF_CONDUCT.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `LICENSE` or `LICENSE.md` or `LICENCE` or `LICENCE.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `nextflow_schema.json`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `nextflow.config`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `README.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/.dockstore.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/CONTRIBUTING.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/ISSUE_TEMPLATE/bug_report.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/ISSUE_TEMPLATE/config.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/ISSUE_TEMPLATE/feature_request.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/markdownlint.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/PULL_REQUEST_TEMPLATE.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/workflows/branch.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/workflows/ci.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/workflows/linting_comment.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/workflows/linting.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `assets/email_template.html`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `assets/email_template.txt`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `assets/sendmail_template.txt`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `assets/nf-core-smrnaseq_logo.png`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `bin/markdown_to_html.py`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `docs/images/nf-core-smrnaseq_logo.png`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `docs/output.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `docs/README.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `docs/README.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `docs/usage.md`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `lib/nfcore_external_java_deps.jar`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `lib/NfcoreSchema.groovy`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `main.nf`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `environment.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `Dockerfile`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `conf/base.config`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/workflows/awstest.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File found: `.github/workflows/awsfulltest.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File not found check: `Singularity`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File not found check: `parameters.settings.json`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File not found check: `bin/markdown_to_html.r`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File not found check: `conf/aws.config`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File not found check: `.github/workflows/push_dockerhub.yml`
* [files_exist](https://nf-co.re/tools-docs/lint_tests/files_exist.html) - File not found check: `.travis.yml`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `manifest.name`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `manifest.nextflowVersion`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `manifest.description`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `manifest.version`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `manifest.homePage`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `timeline.enabled`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `trace.enabled`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `report.enabled`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `dag.enabled`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `process.cpus`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `process.memory`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `process.time`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `params.outdir`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `params.input`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `params.show_hidden_params`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `params.schema_ignore_params`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `manifest.mainScript`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `timeline.file`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `trace.file`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `report.file`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `dag.file`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable found: `process.container`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable (correctly) not found: `params.version`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable (correctly) not found: `params.nf_required_version`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable (correctly) not found: `params.container`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable (correctly) not found: `params.singleEnd`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable (correctly) not found: `params.igenomesIgnore`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable (correctly) not found: `params.name`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config ``timeline.enabled`` had correct value: ``true``
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config ``report.enabled`` had correct value: ``true``
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config ``trace.enabled`` had correct value: ``true``
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config ``dag.enabled`` had correct value: ``true``
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config ``manifest.name`` began with ``nf-core/``
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable ``manifest.homePage`` began with https://github.com/nf-core/
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config ``dag.file`` ended with ``.svg``
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config variable ``manifest.nextflowVersion`` started with >= or !>=
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config ``process.container`` looks correct: ``nfcore/smrnaseq:1.1.0``
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config `params.custom_config_version` is set to `master`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Config `params.custom_config_base` is set to `https://raw.githubusercontent.com/nf-core/configs/master`
* [nextflow_config](https://nf-co.re/tools-docs/lint_tests/nextflow_config.html) - Lines for loading custom profiles found
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.input`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.input_paths`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.outdir`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.protocol`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.genome`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.mirna_gtf`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.fasta`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.bt_index`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.mirtrace_species`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.mirtrace_protocol`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.references_parsed`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.mature`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.hairpin`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.clip_r1`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.three_prime_clip_r1`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.three_prime_adapter`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.min_length`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.skip_qc`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.skip_fastqc`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.skip_multiqc`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.skip_mirdeep`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.save_reference`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.seq_center`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.trim_galore_max_length`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.multiqc_config`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.email`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.email_on_fail`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.max_multiqc_email_size`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.plaintext_email`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.monochrome_logs`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.help`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.tracedir`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.hostnames`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.config_profile_description`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.config_profile_contact`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.config_profile_url`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.publish_dir_mode`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.validate_params`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.max_memory`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.max_cpus`
* [params_used](https://nf-co.re/tools-docs/lint_tests/params_used.html) - Config variable found in `main.nf`: `params.max_time`
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.gitattributes` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `CODE_OF_CONDUCT.md` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `LICENSE` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/.dockstore.yml` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/CONTRIBUTING.md` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/ISSUE_TEMPLATE/bug_report.md` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/ISSUE_TEMPLATE/config.yml` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/ISSUE_TEMPLATE/feature_request.md` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/markdownlint.yml` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/PULL_REQUEST_TEMPLATE.md` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/workflows/branch.yml` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/workflows/linting_comment.yml` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/workflows/linting.yml` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `assets/email_template.html` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `assets/email_template.txt` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `assets/sendmail_template.txt` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `assets/nf-core-smrnaseq_logo.png` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `bin/markdown_to_html.py` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `docs/images/nf-core-smrnaseq_logo.png` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `docs/README.md` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `lib/nfcore_external_java_deps.jar` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `lib/NfcoreSchema.groovy` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.gitignore` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/workflows/push_dockerhub_dev.yml` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `.github/workflows/push_dockerhub_release.yml` matches the template
* [files_unchanged](https://nf-co.re/tools-docs/lint_tests/files_unchanged.html) - `assets/multiqc_config.yaml` matches the template
* [actions_ci](https://nf-co.re/tools-docs/lint_tests/actions_ci.html) - '.github/workflows/ci.yml' is triggered on expected events
* [actions_ci](https://nf-co.re/tools-docs/lint_tests/actions_ci.html) - CI is building the correct docker image: `docker build --no-cache . -t nfcore/smrnaseq:1.1.0`
* [actions_ci](https://nf-co.re/tools-docs/lint_tests/actions_ci.html) - CI is pulling the correct docker image: docker pull nfcore/smrnaseq:dev
* [actions_ci](https://nf-co.re/tools-docs/lint_tests/actions_ci.html) - CI is tagging docker image correctly: docker tag nfcore/smrnaseq:dev nfcore/smrnaseq:1.1.0
* [actions_ci](https://nf-co.re/tools-docs/lint_tests/actions_ci.html) - '.github/workflows/ci.yml' checks minimum NF version
* [actions_awstest](https://nf-co.re/tools-docs/lint_tests/actions_awstest.html) - '.github/workflows/awstest.yml' is triggered correctly
* [actions_awsfulltest](https://nf-co.re/tools-docs/lint_tests/actions_awsfulltest.html) - `.github/workflows/awsfulltest.yml` is triggered correctly
* [readme](https://nf-co.re/tools-docs/lint_tests/readme.html) - README Nextflow minimum version badge matched config. Badge: `20.04.0`, Config: `20.04.0`
* [readme](https://nf-co.re/tools-docs/lint_tests/readme.html) - README had a bioconda badge
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda environment name was correct (nf-core-smrnaseq-1.1.0)
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `conda-forge::r-base=3.6.3`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `conda-forge::r-statmod=1.4.32`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `conda-forge::r-data.table=1.12.4`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `conda-forge::r-gplots=3.0.1.1`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `conda-forge::r-r.methodss3=1.7.1`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `conda-forge::python=3.7.3`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `conda-forge::markdown=3.1.1`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `conda-forge::pymdown-extensions=6.0`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `conda-forge::pygments=2.6.1`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `conda-forge::pigz=2.3.4`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::fastqc=0.11.9`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda package is the latest available: `bioconda::fastqc=0.11.9`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::trim-galore=0.6.5`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::samtools=1.9`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::bowtie=1.2.3`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::multiqc=1.9`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::mirtop=0.4.23`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda package is the latest available: `bioconda::mirtop=0.4.23`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::seqcluster=1.2.7`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda package is the latest available: `bioconda::seqcluster=1.2.7`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::htseq=0.11.3`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::fastx_toolkit=0.0.14`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda package is the latest available: `bioconda::fastx_toolkit=0.0.14`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::seqkit=0.12.0`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::mirtrace=1.0.1`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda package is the latest available: `bioconda::mirtrace=1.0.1`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::bioconductor-edger=3.26.5`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::bioconductor-limma=3.40.2`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda dep had pinned version number: `bioconda::mirdeep2=2.0.1.2`
* [conda_env_yaml](https://nf-co.re/tools-docs/lint_tests/conda_env_yaml.html) - Conda package is the latest available: `bioconda::mirdeep2=2.0.1.2`
* [conda_dockerfile](https://nf-co.re/tools-docs/lint_tests/conda_dockerfile.html) - Found all expected strings in Dockerfile file
* [pipeline_name_conventions](https://nf-co.re/tools-docs/lint_tests/pipeline_name_conventions.html) - Name adheres to nf-core convention
* [template_strings](https://nf-co.re/tools-docs/lint_tests/template_strings.html) - Did not find any Jinja template strings (54 files)
* [schema_lint](https://nf-co.re/tools-docs/lint_tests/schema_lint.html) - Schema lint passed
* [schema_lint](https://nf-co.re/tools-docs/lint_tests/schema_lint.html) - Schema title + description lint passed
* [schema_params](https://nf-co.re/tools-docs/lint_tests/schema_params.html) - Schema matched params returned from nextflow config
* [actions_schema_validation](https://nf-co.re/tools-docs/lint_tests/actions_schema_validation.html) - Workflow validation passed: awsfulltest.yml
* [actions_schema_validation](https://nf-co.re/tools-docs/lint_tests/actions_schema_validation.html) - Workflow validation passed: branch.yml
* [actions_schema_validation](https://nf-co.re/tools-docs/lint_tests/actions_schema_validation.html) - Workflow validation passed: awstest.yml
* [actions_schema_validation](https://nf-co.re/tools-docs/lint_tests/actions_schema_validation.html) - Workflow validation passed: push_dockerhub_dev.yml
* [actions_schema_validation](https://nf-co.re/tools-docs/lint_tests/actions_schema_validation.html) - Workflow validation passed: linting.yml
* [actions_schema_validation](https://nf-co.re/tools-docs/lint_tests/actions_schema_validation.html) - Workflow validation passed: push_dockerhub_release.yml
* [actions_schema_validation](https://nf-co.re/tools-docs/lint_tests/actions_schema_validation.html) - Workflow validation passed: linting_comment.yml
* [actions_schema_validation](https://nf-co.re/tools-docs/lint_tests/actions_schema_validation.html) - Workflow validation passed: ci.yml
* [merge_markers](https://nf-co.re/tools-docs/lint_tests/merge_markers.html) - No merge markers found in pipeline files

### Run details

* nf-core/tools version 1.14
* Run at `2021-05-12 05:30:50`

</details>

ewels commented:
So might be nice to look into upgrading some of these packages if possible. And a full size AWS test is always nice.

> * no default value for `single_end` parameter (we could also remove that completely as not supported?)

Yes we should remove it completely. Is it just this instance?

https://github.com/nf-core/smrnaseq/blob/0b54f7bdd85c75dcfd58e710b627167f75c8d731/main.nf#L184

KevinMenden commented:
Great, I'll try updating some of the packages. I probably won't touch the R stuff for now though because that always breaks 

Yes I think it was only that ocurrence, fixed it here now:
https://github.com/nf-core/smrnaseq/pull/86

Will also use that PR to try around with updating the package versions.

ggabernet commented on .github/workflows/awsfulltest.yml:
```diff
@@ -0,0 +1,43 @@
+            --job-name nf-core-smrnaseq \
+            --job-queue $AWS_JOB_QUEUE \
+            --job-definition $AWS_JOB_DEFINITION \
+            --container-overrides '{"command": ["nf-core/smrnaseq", "-r '"${GITHUB_SHA}"' -profile test --outdir s3://'"${AWS_S3_BUCKET}"'/smrnaseq/results-'"${GITHUB_SHA}"' -w s3://'"${AWS_S3_BUCKET}"'/smrnaseq/work-'"${GITHUB_SHA}"' -with-tower"], "environment": [{"name": "TOWER_ACCESS_TOKEN", "value": "'"$TOWER_ACCESS_TOKEN"'"}]}'
```
```suggestion
            --container-overrides '{"command": ["nf-core/smrnaseq", "-r '"${GITHUB_SHA}"' -profile test_full --outdir s3://'"${AWS_S3_BUCKET}"'/smrnaseq/results-'"${GITHUB_SHA}"' -w s3://'"${AWS_S3_BUCKET}"'/smrnaseq/work-'"${GITHUB_SHA}"' -with-tower"], "environment": [{"name": "TOWER_ACCESS_TOKEN", "value": "'"$TOWER_ACCESS_TOKEN"'"}]}'
```

ggabernet replied:
as you provided the tests data in test_full this can be updated :)

KevinMenden replied:
Ah, thanks! 👍

ggabernet reviewed (approved):
It looks good to me from the pipeline structure perspective, but have no idea on smRNAseq analysis 😄

ewels commented:
Linting shows this:

```
Config variable not found in main.nf: params.gtf
```

Don't know if that needs checking before release?

KevinMenden commented:
Thanks both for having a look! Will see what's going on with that `params.gtf` later 👍

KevinMenden commented:
Turns out the `gtf` param is indeed not used at all, we only have a `mirna_gtf` param that we use. So I removed `gtf` and replaced it by `mirna_gtf` here: https://github.com/nf-core/smrnaseq/pull/91
